docs/packaging-your-sql-project/intro-to-python-scripting-and-pipelines.md

```diff
@@ -11,4 +11,143 @@ kernelspec:
   name: python3
 ---
 
-# Introduction to Python scripting and data pipelines
\ No newline at end of file
+# Introduction to Python scripting and data pipelines
+
+In this section of the course, we will delve into the world of Python scripting and explore how it plays a crucial role in building data pipelines. You'll learn about `.py` scripts and good software development practices for scripting. Through the power of Python scripting, you will gain insights into data pipelines and how they automate data ingestion tasks, providing you with the tools to create robust and efficient data workflows.
+
+## `.py` script vs Jupyter Notebook
+
+A `.py` script is a standalone file that contains Python code, which can be written with the help of text editors or IDE's (integrated development environment).
+
+Unlike Jupyter notebooks, which are interactive environments for code execution and documentation, `.py` scripts are designed for running Python code in a more traditional, non-interactive manner. They are commonly used for writing programs that can be executed from the command line, as shown below, or integrated into software projects due to [modularization](https://ploomber-sql.readthedocs.io/en/latest/packaging-your-sql-project/intro-to-python-scripting-and-pipelines#modularization) (variables and functions inside a Python script can be imported from another script).
+
+After writing and saving a Python script, you can run it by executing the `python` command on your terminal, like so:
+
+```bash
+python my_script.py
+```
+
+<b>Note:</b> The command above assumes that your terminal is currently inside the folder where the `my_script.py` file is located.
+
+IDE's, such as [Visual Studio Code](https://code.visualstudio.com/) and [Pycharm](https://www.jetbrains.com/pycharm/) on the other hand, offer a more interactive way of writing and running Python scripts. They provide features such as syntax highlighting, code completion, and debugging tools, making it easier to write and test your code.
+
+```important
+`.py` scripts are always executed linearly from top to bottom, unlike Jupyter notebooks, where you can run cells in any order. This makes debugging easier in `.py` scripts, as you can easily trace the flow of execution.
+```
+
+Jupyter Notebooks are loved by Data Scientists for their interactive nature. Outputs, such as messages, plots, and dataframes, appear under each cell upon execution, and look great out-of-the-box. Moreover, Jupyter Notebooks are great for data analyses and sharing results with colleagues. However, they must be served and accessed through a web browser, making them slightly harder to use than scripts.
```

Comment:
Analyses -> analysis

Unless you meant plural? It sounds strange but open to hearing rationale

Reply:
Yes analysis should be the correct term here, fixed in latest commit